view/frontend/templates/button_product_page.phtml

```diff
@@ -0,0 +1,278 @@
+                        location.href = callbacks.success_url;
+                    }
+                },
+
+                onCheckoutStart: function() {
+                    trackCallbacks.onCheckoutStart();
+                },
+
+                onShippingDetailsComplete: function() {
+                    trackCallbacks.onShippingDetailsComplete();
+                },
+
+                onShippingOptionsComplete: function() {
+                    trackCallbacks.onShippingOptionsComplete();
+                },
+
+                onPaymentSubmit: function() {
+                    trackCallbacks.onPaymentSubmit();
+                },
+
+                success: function (transaction, callback) {
+                    // abort previously sent save order request.
+                    if (save_request) save_request.abort();
+                    // get thr transaction reference
+                    var parameters = 'reference='+transaction.reference;
+                    // update order ajax request callback
+                    // sets the success order page redirect url from received data
+                    // and calls the final Bolt defined callback
+                    var onSuccess = function(data){
+                        if (data.status !== 'success') {
+                            if (data.message) {
+                                showBoltErrorMessage();
+                                // pretend order creation was success...
+                                // we need to call this; otherwise bolt modal show infinte spinner.
+                                callback();
+                            }
+
+                            return false;
+                        }
+                        try {
+                            <?php echo $block->getjavascriptSuccess(); ?>
+                            callbacks.success_url = data.success_url;
+
+                            trackCallbacks.onSuccess();
+                        } finally {
+                            callback();
+                        }
+                    };
+                    // ajax call to the update order transaction data endpoint.
+                    // passes the bolt transaction reference
+                    save_request = ajaxPost(settings.save_order_url, parameters, onSuccess);
+                },
+
+                check: function () {
+                    /**
+                     * On Bolt button click check if guest checkout is allowed.
+                     * Display login popup to guest customers if it is not. The
+                     * Magento customerData and authenticationPopup objects are
+                     * used.
+                     */
+
+                    //check if we have inventory on stock
+                    if ( (max_qty>=0) && $qty && (Number($qty.val()) > max_qty)) {
+                        window.alert("Please correct the quantity. The most you may purchase is " + max_qty + ".");
```

Comment:
@vitaliyreznikov Is this the same error message we display in WooC? I want to make sure we are consistent.

Reply:
I believe this error should be the same as on native m2 error when we try to add to cart more inventory that we have.
Now it's not the case, let me fix it on separate PR https://app.asana.com/0/0/1156147394609521/f
For now, I only changed the alert text